From d5e14b3a9aaf5d78b95e159f3c936410c7f15b8b Mon Sep 17 00:00:00 2001
From: ExtremeXT <extremextdev@gmail.com>
Date: Sat, 6 Sep 2025 14:03:36 +0300
Subject: [PATCH 2/2] aidl: usb: gadget: Don't bail out on mass_storage link
 fail

* This feature requires a change in the kernel config and another one in the USB init:
* https://github.com/ExtremeXT/990_upstream_v2/commit/e1e4a193d4ee5b4f2b439c7821c5767116f8a402
* https://github.com/ExtremeXT/android_device_samsung_universal9830-common/commit/bdd598c4d0c569fc3aa4a7a2fa679bf34498240d

* This is not a vital feature, so there is no reason to bail out and completely kill USB when it fails.

Change-Id: I7e5a391d100ef77b9904384119e53629dc69260b
Signed-off-by: ExtremeXT <extremextdev@gmail.com>
---
 aidl/usb/gadget/UsbGadget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aidl/usb/gadget/UsbGadget.cpp b/aidl/usb/gadget/UsbGadget.cpp
index 8bfefe0..7eb91fc 100644
--- a/aidl/usb/gadget/UsbGadget.cpp
+++ b/aidl/usb/gadget/UsbGadget.cpp
@@ -273,7 +273,7 @@ Status UsbGadget::setupFunctions(long functions, const shared_ptr<IUsbGadgetCall
     if ((functions == GadgetFunction::ADB) || ((functions & GadgetFunction::MTP) != 0)) {
         ALOGI("setCurrentUsbFunctions mass_storage");
         if (linkFunction("mass_storage.0", i++))
-            return Status::ERROR;
+            ALOGW("%s: Failed to link mass_storage", __FUNCTION__);
     }
 
     // Pull up the gadget right away when there are no ffs functions.
-- 
2.54.0

